src/main/java/ch/njol/skript/conditions/CondIsEnchanted.java

```diff
@@ -33,50 +31,62 @@
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.skript.util.EnchantmentType;
 import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Is Enchanted")
 @Description("Checks whether an item is enchanted.")
 @Examples({"tool of the player is enchanted with efficiency 2",
-		"helm, chestplate, leggings or boots are enchanted"})
-@Since("1.4.6")
+		"helm, chestplate, leggings or boots are enchanted",
+		"",
+		"# For enchanted books",
+		"player's tool is stored enchanted",
+		"event-item is enchanted with stored power 2"})
+@Since("1.4.6, INSERT VERSION (stored enchantments)")
 public class CondIsEnchanted extends Condition {
 	
 	static {
-		PropertyCondition.register(CondIsEnchanted.class, "enchanted [with %-enchantmenttype%]", "itemtypes");
+		Skript.registerCondition(CondIsEnchanted.class,
+			"%itemtypes% ((is|are)|not:(isn't|is not|aren't|are not)) enchanted [with %-enchantmenttypes%]",
+			"%itemstypes% ((has|have)|not:(doesn't have|don't have)) %-enchantmenttypes% stored",
+			"%-enchantmenttypes% ((is|are)|not:(is not|isn't|are not|aren't)) stored on %itemtypes%"
```

Comment:
Why is %-enchantmenttypes% set but it's not surrounded by an optional? It should be the optional to signify that it's optional.

I also don't agree with this being nullable optional. This is an old syntax yes, I understand, but we did the same with the weather condition `is raining` it's designed to be a singleton condition used in the context of the event and default variables. This however does not want a default variable, so I see no point in it being nullable.